controllers/machinedeletionremediation_controller.go

```diff
@@ -201,3 +207,24 @@ func extractNameAndNamespace(nameNamespace string, nodeName string) (string, str
 	}
 	return "", "", fmt.Errorf(invalidValueMachineAnnotationError, nodeName)
 }
+
+func getMachineStatusPhase(machine *unstructured.Unstructured) (string, error) {
+	phase := "unknown"
+
+	status, ok, err := unstructured.NestedMap(machine.Object, "status")
+	if err != nil {
+		return phase, fmt.Errorf("could not get Machine's status: error %v", err)
+	}
+	if !ok {
+		return phase, fmt.Errorf("Machine object does not have a status field")
+	}
+
+	phase, ok, err = unstructured.NestedString(status, "phase")
+	if err != nil {
+		return phase, fmt.Errorf("could not get Machine's status.phase: error %v", err)
```

Comment:
Nit: I think wrapping the original error would be more correct for this use case